utils/_context/virtual_machines.py

```diff
@@ -188,6 +175,19 @@ def start(self):
     def set_ip(self, instance_ip):
         self.ip = instance_ip
 
+    def set_components_json(self, components_json):
```

Comment:
Moving `get_component` logic inside `set_components_json`, rename it `set_components(self, json_source)`, and get result using `TestedVirtualMachine.components`.